src/Common/LRUCache.h

```diff
@@ -57,23 +76,38 @@ class LRUCache
         return res;
     }
 
-    void set(const Key & key, const MappedPtr & mapped)
+    /**
+     * set() will fail (return false) if there is no space left and no keys could be evicted.
+     * Eviction permission of each key is defined by EvictPolicy. In default policy there is no restriction.
+     */
+    bool set(const Key & key, const MappedPtr & mapped)
```

Comment:
Looks incorrect, it should be `void set(...)` and `bool trySet(...)` instead.
The first must throw if failed to set, second just returns false if failed to set.

Reply:
hm, well lets leave as is, no change needed.

Reply:
just noticed you already fixed..

Reply:
No need to throw an exception on `set()` ?

Reply:
No because previously there also was no exception, and we need to preserve previous behavior. So now I though because of this there is no need to separate into `set` and `trySet` -- because previously `set` did not guarantee insertion. Separation could make sense if `set` guaranteed insertion. Also I think may be put this part with lru cache changes also in a separate pr, so that someone apart from me will also take a look and we will merge it earlier than other changes.

Reply:
have added a new pr #33098